moshi/moshi/models/tts.py

```diff
@@ -676,3 +678,109 @@ def get_prefix(self, audio_path: Path) -> torch.Tensor:
         null_text = torch.full_like(prefix[:1], self.machine.token_ids.zero)
         prefix = torch.cat([null_text, prefix], dim=0)
         return prefix
+
+    def simple_generate(
+        self, text: str | list[str], voice: str | list[str], show_progress: bool = True
+    ) -> list[torch.Tensor]:
+        """Generate audio directly from text and voice name(s).
+
+        `text` and `voice` can be "broadcast" against each other like in NumPy: if one
+        is a list and the other a single item, the single item is repeated for each item
+        in the list. If both are lists, they must have the same length and voice[i]
+        will be used for text[i].
+
+        If you need more control, refer to the examples in
+        https://github.com/kyutai-labs/delayed-streams-modeling
+
+        Args:
+            text: text or list of texts to synthesize.
+            voice: voice name or list of voice names to use. Each entry can be a path
+                relative to the voice repo (see https://huggingface.co/kyutai/tts-voices)
+                or a path to a local .safetensors file with the voice embedding.
+            show_progress: whether to show a progress bar.
+
+        Returns:
+            list of 1D float32 tensors with the generated audio, sampled at `self.mimi.sample_rate`.
+            It's a list because the length of the generated audio can vary.
+        """
+        multiple_texts = isinstance(text, list)
+        multiple_voices = isinstance(voice, list)
+
+        texts: list[str]
+        voices: list[str]
+        if not multiple_texts and not multiple_voices:
+            texts = [text]
+            voices = [voice]
+        elif multiple_texts and not multiple_voices:
+            texts = text
+            voices = [voice] * len(text)
+        elif not multiple_texts and multiple_voices:
+            texts = [text] * len(voice)
+            voices = voice
+        elif multiple_texts and multiple_voices:
+            if len(text) != len(voice):
+                raise ValueError(
+                    f"Number of texts and voices must match, got {len(text)=} != {len(voice)=}"
+                )
+            if len(text) == 0:
+                raise ValueError("Got empty list, nothing to generate")
+            texts = text
+            voices = voice
+        else:
+            assert False, "Never happens, just making the type-checker happy"
+
+        entries_batch = [self.prepare_script([t], padding_between=1) for t in texts]
+        voice_paths = [
+            Path(v) if v.endswith(".safetensors") else self.get_voice_path(v)
+            for v in voices
+        ]
+
+        condition_attributes_batch = [
+            self.make_condition_attributes([vp], cfg_coef=2.0) for vp in voice_paths
```

Comment:
cfg_coef could be a param no?

this code will not work for the public model https://huggingface.co/kyutai/tts-0.75b-en-public which uses prefix audio for speaker conditioning, and does not have cfg distillation. I put the code for using the public model in the HF readme in case you want to also support it